irohad/ordering/on_demand_os_transport.hpp

```diff
@@ -74,9 +74,11 @@ namespace iroha {
 
         /**
          * Callback on receiving transactions
+         * @param round - expected proposal round
          * @param transactions - vector of passed transactions
          */
-        virtual void onTransactions(CollectionType transactions) = 0;
+        virtual void onTransactions(RoundType round,
```

Comment:
can it take const ref as argument?

Reply:
Not sure about the interface. Making it a ref-to-const will require maintaining object lifetime outside the function. Such behavior is harder to maintain in case of callbacks.